File: Is_Subsequence/result.py
# Parameters involved are the following: 
#   s - String s containing some letters 
#   t - String t containing some letter 
# This function compares to see if the String t contains a subsequence of 
# String s in the same order of characters as it appears in String s.

def isSubsequence(s, t):
    s_length = len(s);
    t_length = len(t);
    indexfound = 0
    found = False 
    for i in range(s_length):
        found = False # Every character inside s would begin with a false 

        for j in range(indexfound, t_length):
            if (t[j] == s[i]): 
                indexfound = j + 1
                found = True
                
            
        # If we go through all of the characters and can't find a matching
        # character in the order we expected it to be then we will have to
        # return false midway 
        if (found != True):
            return False

    if (found != True):
        return False
    else:
        return True

s = "abc"  
t = "ahbgdc"
print(isSubsequence(s, t))          

File: Is_Subsequence/result.java
import java.util.List;
import java.util.ArrayList;

class Solution{
	public static boolean isSubsequence(String s, String t){
		int count = 0;
		int indexfound = 0;
		boolean found = false; 
		for (int i = 0 ; i < s.length(); i ++){
			for (int j = i; j < t.length(); j ++){
				if ((t.charAt(j) == s.charAt(i)) && (j >= indexfound)){
					indexfound = j;
					count++;
					break;
				}	
			}		
		}
		
		if (count == s.length()){
			return true;
		}
		return false;
	}

	public static void main(String[] args){
		String s = "abc";
		String t = "ahbgdc";
		boolean output = isSubsequence(s, t);
		System.out.println(output);
	}
}
